### PR TITLE
confirm modal for delete node

### PR DESCRIPTION
**Pull request by rohitcmu**
https://issues.cask.co/browse/CDAP-8972

### CODE REVIEW

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 1]
```diff
@@ -605,8 +605,19 @@ angular.module(PKG.name + '.commons')
...
     vm.onNodeDelete = function (event, node) {
       event.stopPropagation();
-      DAGPlusPlusNodesActionsFactory.removeNode(node.name);
-      vm.instance.remove(node.name);
+      let modal = $uibModal.open({
+        templateUrl: 'dag-plus/confirm-deletion-modal.html',
+        size: 'sm',
+        backdrop: true,
+        keyboard: true,
+        windowTopClass: 'delete-modal',
```
Comment (elfenheart) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js:613]: this class should be `confirm-modal hydrator-modal center`. Then you can wrap the buttons on the footer with another div with `pull-right`.

see `/templates/create/popovers/run-preview-action-confirmation-modal.html`

**Comment by leacd** [comment]
A few style and text changes.
image

Modal header is gray #eeeeee
Modal has a 20 px margin
Font is Helvetica reg 18 px

**Comment by nitinmotgi** [comment]
This PR should not be merged. We should add undo immediately. @bdmogal !!! WHY ?

**Comment by bdmogal** [comment]
@nitinmotgi I think confirmation is unrelated to undo. IMO, We should do undo/redo apart from this (its not currently planned for this release). If I have made some changes, and I click the cross, I think there should be a confirmation dialog.

However, we shouldn't see this, if there haven't been any changes to the node config. e.g. if I just drop a node and delete it right away, I shouldn't see a confirmation dialog. @rohitcmu can you confirm that that's the case?

Let's discuss in person?

**Comment by bdmogal** [comment]
Closing this per discussion with @nitinmotgi. We do not want to show a user a confirmation screen in the studio, when he is in active pipeline development. Instead, we will implement undo later.

**Pull request closed without merging**